src/scorecard.js

```diff
@@ -0,0 +1,90 @@
+class Scorecard {
+  constructor() {
+    this._pinsKnocked = []
+    this._frame = 0
+    this._roll = 0
```

Comment:
The fact that you need to keep track of which roll in the frame and which frame in the game point to me towards the idea of extracting Frame functionality into its own class, that could keep its own pins and scores.

Then the game is responsible for creating a new frame when needed. The frame could tell it if it's finished, if it still needs some bonuses, etc.